centos-upgrade-first-stage.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+#
+# Script to run the first stage (everything through rebooting to flip into
+# the new kernel) for upgrading the CentOS AMIs from the imported version
+# to the latest in that major release series.
+#
+
+# Upgrade everything!
+sudo yum -y upgrade
+
+if test `cat /etc/centos-release | sed -n 's/CentOS release \([0-9]\).*/\1/p'` == 6; then
+    # Remove the 2.6.32 kernel series.
+    sudo yum -y remove kernel kernel-devel
+    export NEXT=`ls -1t /lib/modules/ | head -n 1`
```

Comment:
I would prefer `ls -1v` and take last one. I am not sure about sort by time `t`

Reply:
-t is actually the best option; you don't want the highest kernel version, you want the last one that was installed (because that's what Grub is going to boot).  But even -t is probably a bit too fragile.  I'll think of a better option here.  Now that I just made it a shell script (instead of inline packer commands), I think grepping through grub.cfg may (unfortunately) be the best option.  The other option is to only run this on non-ENA platforms, which would make this code much simpler.